src/mixins/helpers.js

```diff
@@ -53,26 +53,26 @@ var helpers = {
       this.autoPlay(); // once we're set up, trigger the initial autoplay.
     });
   },
-  update: function (props) {
+  update: function (props, recursionLevel=0) {
     const slickList = ReactDOM.findDOMNode(this.list);
     // This method has mostly same code as initialize method.
     // Refactor it
     var slideCount = React.Children.count(props.children);
-    var listWidth = this.getWidth(slickList);
-    var trackWidth = this.getWidth(ReactDOM.findDOMNode(this.track));
+    var listWidth = getWidth(slickList);
+    var trackWidth = getWidth(ReactDOM.findDOMNode(this.track));
```

Comment:
getWidth(slickTrack);

Reply:
@developer-lindner this looks great. I'll try and change that.
I encourage you to create a PR for dev branch with such ideas.